cfy_manager/components/restservice/restservice.py

```diff
@@ -298,35 +297,91 @@ def _validate_admin_password_and_security_config(self):
-            self._make_paths()
-            self._configure_restservice()
-            self._enter_sanity_mode()
-            self._configure_db()
-            systemd.configure(RESTSERVICE)
-            systemd.restart(RESTSERVICE)
-            if config[CLUSTER_JOIN]:
-                logger.info('Extra node in cluster, will verify rest-service '
-                            'after clustering configured')
-            else:
-                self._verify_restservice_alive()
-        finally:
-            self._exit_sanity_mode()
+    def _wait_for_haproxy_startup(self):
+        logger.info('Waiting for DB proxy startup to complete...')
+        healthy = False
+        for attempt in range(60):
+            # Get the haproxy status data
+            haproxy_csv = requests.get(
+                'http://localhost:7000/admin?stats;csv;norefresh'
+            ).text
+
+            # Example output (# noqas are not part of actual output):
+            # # pxname,svname,qcur,qmax,scur,smax,slim,stot,bin,bout,dreq,dresp,ereq,econ,eresp,wretr,wredis,status,weight,act,bck,chkfail,chkdown,lastchg,downtime,qlimit,pid,iid,sid,throttle,lbtot,tracked,type,rate,rate_lim,rate_max,check_status,check_code,check_duration,hrsp_1xx,hrsp_2xx,hrsp_3xx,hrsp_4xx,hrsp_5xx,hrsp_other,hanafail,req_rate,req_rate_max,req_tot,cli_abrt,srv_abrt,comp_in,comp_out,comp_byp,comp_rsp,lastsess,last_chk,last_agt,qtime,ctime,rtime,ttime,  # noqa
+            # stats,FRONTEND,,,1,1,2000,7,553,83778,0,0,0,,,,,OPEN,,,,,,,,,1,1,0,,,,0,1,0,1,,,,0,6,0,0,0,0,,1,1,7,,,0,0,0,0,,,,,,,,  # noqa
+            # stats,BACKEND,0,0,0,0,200,0,553,83778,0,0,,0,0,0,0,UP,0,0,0,,0,89,0,,1,1,0,,0,,1,0,,0,,,,0,0,0,0,0,0,,,,,0,0,0,0,0,0,0,,,0,0,0,0,  # noqa
+            # postgres,FRONTEND,,,0,0,2000,0,0,0,0,0,0,,,,,OPEN,,,,,,,,,1,2,0,,,,0,0,0,0,,,,,,,,,,,0,0,0,,,0,0,0,0,,,,,,,,  # noqa
+            # postgres,postgresql_192.0.2.46_5432,0,0,0,0,100,0,0,0,,0,,0,0,0,0,DOWN,1,1,0,1,1,89,89,,1,2,1,,0,,2,0,,0,L7STS,503,3,,,,,,,0,,,,0,0,,,,,-1,HTTP status check returned code <503>,,0,0,0,0,  # noqa
+            # postgres,postgresql_192.0.2.47_5432,0,0,0,0,100,0,0,0,,0,,0,0,0,0,UP,1,1,0,0,0,89,0,,1,2,2,,0,,2,0,,0,L7OK,200,3,,,,,,,0,,,,0,0,,,,,-1,HTTP status check returned code <200>,,0,0,0,0,  # noqa
+            # postgres,postgresql_192.0.2.48_5432,0,0,0,0,100,0,0,0,,0,,0,0,0,0,DOWN,1,1,0,1,1,87,87,,1,2,3,,0,,2,0,,0,L7STS,503,2,,,,,,,0,,,,0,0,,,,,-1,HTTP status check returned code <503>,,0,0,0,0,  # noqa
+            # postgres,BACKEND,0,0,0,0,200,0,0,0,0,0,,0,0,0,0,UP,1,1,0,,0,89,0,,1,2,0,,0,,1,0,,0,,,,,,,,,,,,,,0,0,0,0,0,0,-1,,,0,0,0,0,  # noqa
+            haproxy_data = list(csv.reader(
+                haproxy_csv.lstrip('# ').splitlines()
+            ))
+            headers = haproxy_data[0]
+            haproxy_status = [
+                {
+                    headers[i]: row[i]
+                    for i in range(len(headers))
+                }
+                for row in haproxy_data
+            ]
```

Comment:
this stuff should be able to be replaced with csv.DictReader

Reply:
Yup, much nicer, thanks!